### PR TITLE
Alerting: Add error to annotations when execution error state is Alerting

### PR DESCRIPTION
**Pull request by alexander-akhmetov**
**What is this feature?**

Add error to annotations when execution error state is Alerting

**Why do we need this feature?**

Currently we do not add anything to the annotations, so when the rule is configured to be `Alerting` on errors, there is no additional information about the error in the alert notifications.

**Without the change**
image

**With the change**
New `Error`, `datasource_uid` and `ref_id` annotations.
image

<details>
  <summary><b>Grafana UI</b></summary>

### Alerting

image

### Back to normal

image

</details>

### CODE REVIEW

File: pkg/services/ngalert/state/manager.go [hunk 1]
```diff
@@ -463,7 +463,7 @@ func (st *Manager) setNextState(ctx context.Context, alertRule *ngModels.AlertRu
 	//
 	// This is temporary change to make sure that the labels are not persistent in the state after it was in Error state
 	// TODO yuri. Remove it when correct Error result with labels is provided
-	if currentState.State == eval.Error && result.State != eval.Error {
+	if (currentState.State == eval.Error || currentState.State == eval.Alerting) && result.State != eval.Error {
```
Comment (alexander-akhmetov) [inline pkg/services/ngalert/state/manager.go:466]: Maybe with `currentState.State == eval.Alerting` we should also check that `rule.ExecErrState == models.AlertingErrState`? Not sure

File: pkg/services/ngalert/state/manager_private_test.go [hunk 2]
```diff
@@ -3689,14 +3689,17 @@ func TestProcessEvalResults_StateTransitions(t *testing.T) {
 				},
 			},
 			{
-				desc: "t1[QueryError] t2[{}:normal] at t2",
+				desc: "t1[QueryError] t2[{}:normal] at t2,t3",
```
Comment (yuri-tceretian) [inline pkg/services/ngalert/state/manager_private_test.go:3692]: please change description to "t1[QueryError] t2[{}:normal] t3[{}:normal] at t2,t3",

Comment (alexander-akhmetov) [reply pkg/services/ngalert/state/manager_private_test.go]: updated!

**Review by yuri-tceretian — CHANGES_REQUESTED** [review]
when Error is mapped to either Alerting or Normal, no new labels should be added to the state. Otherwise, we break notification pipline. `ref_id` and `datasource_uid` should be included into annotations.

**Comment by alexander-akhmetov** [comment]
Updated to use annotations: https://github.com/grafana/grafana/pull/91594/commits/026178ba9310ef9731867fcf2ba6253b9bfedba2

**Review by yuri-tceretian — APPROVED** [review]
LGTM! Great change!

**Pull request merged**